From: Chris Renshaw <osm0sis@outlook.com>
Date: Thu, 22 Jun 2017 14:07:04 -0300
Subject: [PATCH] additional fix-up for IPV6 on Android API 21+

fixes: networking/interface.c:62:8: error: redefinition of 'struct in6_ifreq'

and:
error: field 'ifru_addr' has incomplete type
  struct sockaddr ifru_addr;
error: field 'ifru_dstaddr' has incomplete type
  struct sockaddr ifru_dstaddr;
...
since sockaddr was moved from linux/socket.h to sys/socket.h in newer NDK revisions

---
 networking/interface.c | 2 ++
 networking/route.c     | 1 +
 2 files changed, 3 insertions(+)

diff --git a/networking/interface.c b/networking/interface.c
index c5c8f2cdd..ec9bbf225 100644
--- a/networking/interface.c
+++ b/networking/interface.c
@@ -56,6 +56,7 @@
 
 #ifdef HAVE_AFINET6
 # ifndef _LINUX_IN6_H
+#  ifndef __BIONIC__
 /*
  * This is from linux/include/net/ipv6.h
  */
@@ -64,6 +65,7 @@ struct in6_ifreq {
 	uint32_t ifr6_prefixlen;
 	unsigned int ifr6_ifindex;
 };
+#  endif
 # endif
 #endif /* HAVE_AFINET6 */
 
diff --git a/networking/route.c b/networking/route.c
index 08bea7311..42f8e43f8 100644
--- a/networking/route.c
+++ b/networking/route.c
@@ -43,6 +43,7 @@
 //usage:     "\n	-e	Display other/more information"
 //usage:     "\n	-A inet" IF_FEATURE_IPV6("{6}") "	Select address family"
 
+#include <netinet/in.h>
 #include <net/route.h>
 #include <net/if.h>
 #include <linux/in6.h>
-- 
2.12.3

